chat_app_be/chat_app_be/Controllers/AuthController.cs

```diff
@@ -0,0 +1,108 @@
+﻿using chat_app_be.Dtos;
+using chat_app_be.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Text;
+using chat_app_be.Interfaces;
+using chat_app_be.Models.Auth;
+
+namespace chat_app_be.Controllers
+{
+    [Route("api/")]
+    [ApiController]
+    public class AuthController : ControllerBase
+    {
+        private readonly JwtOptions _jwtOptions;
+        private readonly IUserRepository _userRepository;
+
+        public AuthController(IOptions<JwtOptions> jwtOptions, IUserRepository userRepository)
+        {
+            _jwtOptions = jwtOptions.Value;
+            _userRepository = userRepository;
+        }
+
+        [HttpPost("register")]
+        public async Task<ActionResult<UserRequestDto>> Register(UserDto user)
+        {
+            if (await _userRepository.UserExistsAsync(user.Username))
+            {
+                return BadRequest("Username already exists");
+            }
+
+            CreatePasswordHash(user.Password, out byte[] passwordHash, out byte[] passwordSalt);
+
+            var newUser = new User
+            {
+                Username = user.Username,
+                DisplayName = user.DisplayName,
+                PasswordHash = passwordHash,
+                PasswordSalt = passwordSalt
+            };
+
+            await _userRepository.AddUserAsync(newUser);
+
+            return Ok(user);
```

Comment:
Don't return the `user` object because it includes user secret information such as password, etc. We could return status 200 only for the creation API or another mapped Dto.